test_app/tests/lib/utils/test_validation.py

```
@@ -708,3 +709,137 @@ def test_validate_port_string_edge_cases(self, port_string, expected, descriptio
+
+    @pytest.mark.parametrize(
+        "value,description",
+        [
+            ("\x00 null byte", "null byte"),
+            ("\x08 backspace", "backspace"),
+            ("\x0d bare CR", "bare carriage return"),
+            ("\x1b[31m red", "ANSI escape sequence"),
+            ("\x7f DEL", "DEL character"),
+            ("\x80 C1 control", "C1 control character"),
+            ("\x9f end of C1", "C1 block end"),
+            ("​ zero-width space", "zero-width space"),
```

Ruff (0.16.1)

[error] Invalid unescaped character zero-width-space, use "\u200B" instead

Replace with escape sequence

(PLE2515)